libbeat/dashboards/import_dashboards.go

```diff
@@ -0,0 +1,559 @@
+	"github.com/elastic/beats/libbeat/outputs/outil"
+)
+
+type Options struct {
+	KibanaIndex string
+	ES          string
+	Index       string
+	Dir         string
+	File        string
+	Beat        string
+	Url         string
+	User        string
+	Pass        string
+}
+
+type CommandLine struct {
+	flagSet *flag.FlagSet
+	opt     Options
+}
+
+type Importer struct {
+	cl     *CommandLine
+	client *elasticsearch.Client
+}
+
+func ParseCommandLine() (*CommandLine, error) {
+	var cl CommandLine
+
+	cl.flagSet = flag.NewFlagSet("import", flag.ContinueOnError)
+
+	cl.flagSet.Usage = func() {
+		const usage = `
+Usage: ./import_dashboards [options]
+
+Kibana dashboards are stored in a special index in Elasticsearch together with the searches, visualizations, and indexes that they use.
+
+You can import the dashboards, visualizations, searches, and the index pattern for any Beat:
+  1. from a local directory:
+	./import_dashboards -dir etc/kibana
+  2. from a directory of a local zip archive:
+	./import_dashboards -dir metricbeat -file beats-dashboards-1.2.3.zip
+  3. from a directory of zip archive available online:
+	./import_dashboards -dir metricbeat -url http://download.elastic.co/beats/dashboards/beats-dashboards-1.2.3.zip
+
+Options:
+`
+
+		os.Stderr.WriteString(usage)
+		cl.flagSet.PrintDefaults()
+
+	}
+
+	cl.flagSet.StringVar(&cl.opt.KibanaIndex, "k", ".kibana", "Kibana index")
+	cl.flagSet.StringVar(&cl.opt.ES, "es", "http://127.0.0.1:9200", "Elasticsearch URL")
+	cl.flagSet.StringVar(&cl.opt.User, "user", "", "Username to connect to Elasticsearch")
+	cl.flagSet.StringVar(&cl.opt.Pass, "pass", "", "Password to connect to Elasticsearch")
+	cl.flagSet.StringVar(&cl.opt.Index, "i", "", "Overwrites the Elasticsearch index name. For example you can replaces metricbeat-* with custombeat-*")
+	cl.flagSet.StringVar(&cl.opt.Dir, "dir", "", "Directory containing the subdirectories: dashboard, visualization, search, index-pattern. eg. kibana/")
+	cl.flagSet.StringVar(&cl.opt.File, "file", "", "Zip archive file containing the Beats dashboards.")
+	cl.flagSet.StringVar(&cl.opt.Url, "url", "", "URL to the zip archive containing the Beats dashboards")
+
+	return &cl, nil
+}
+
+func (cl *CommandLine) Read() error {
+
+	if err := cl.flagSet.Parse(os.Args[1:]); err != nil {
+		return err
+	}
+
+	if cl.opt.Url == "" && cl.opt.File == "" && cl.opt.Dir == "" {
+		return errors.New("ERROR: Missing input. Please specify one of the options -file, -url or -dir")
+	}
+
+	return nil
+}
+
+func New() (*Importer, error) {
+	importer := Importer{}
+
+	/* command line */
+	cl, err := ParseCommandLine()
+	if err != nil {
+		cl.flagSet.Usage()
+		return nil, err
+	}
+	err = cl.Read()
+	if err != nil {
+		return nil, err
+	}
+	importer.cl = cl
+
+	/* connect to Elasticsearch */
+	fmtstr, err := fmtstr.CompileEvent(cl.opt.Index)
+	if err != nil {
+		return nil, fmt.Errorf("fail to compile the Elasticsearch index pattern: %s", err)
```

Comment:
I suggest to replace this error message with `fail to build the Kibana index pattern: ...` The reason is that compile sounds to me like a binary is generated and index pattern relates to Kibana and not elasticsearch.